File: algorithms/code/leetcode/lc026_remove_duplicates_from_sorted_array/LC026RemoveDuplicatesFromSortedArray.Tests.cs
using System.Linq;
using Xunit;

namespace AlgorithmDesign.code.leetcode.lc026_remove_duplicates_from_sorted_array
{
    public class SolutionTests
    {
        [Fact]
        public void test_1_2_2_3_4_4_7()
        {
            int[] nums1 = { 1, 2, 2, 3, 4, 4, 7 };
            int length = new Solution().RemoveDuplicates(nums1);
            Assert.Equal(5, length);
            int[] expected = { 1, 2, 3, 4, 7 };
            Assert.Equal(expected, nums1.Take(length));
        }
    }
}

File: algorithms/code/leetcode/lc026_remove_duplicates_from_sorted_array/LC026RemoveDuplicatesFromSortedArrayTests.java
package leetcode.lc026_remove_duplicates_from_sorted_array;

import java.util.Arrays;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;

public final class LC026RemoveDuplicatesFromSortedArrayTests {
    @Test
    public void test1223447() throws Exception {
        LC026RemoveDuplicatesFromSortedArray solution;
        solution = new LC026RemoveDuplicatesFromSortedArray();
        int[] nums1 = {1, 2, 2, 3, 4, 4, 7};
        int length = solution.removeDuplicates(nums1);
        assertEquals(5, length);
        int[] expected = {1, 2, 3, 4, 7};
        assertArrayEquals(expected, Arrays.copyOfRange(nums1, 0, length));
    }
}
